test/integration/wan_lan_dht_test.go

```diff
@@ -0,0 +1,277 @@
+StartupWait:
+	for {
+		select {
+		case err := <-testPeer.DHT.LAN.RefreshRoutingTable():
+			if err != nil {
+				fmt.Printf("Error refreshing routing table: %v\n", err)
+			}
+			if testPeer.DHT.LAN.RoutingTable() == nil ||
+				testPeer.DHT.LAN.RoutingTable().Size() == 0 ||
+				err != nil {
+				time.Sleep(100 * time.Millisecond)
+				continue
+			}
+			break StartupWait
+		case <-startupCtx.Done():
+			startupCancel()
+			return fmt.Errorf("expected faster dht bootstrap")
+		}
+	}
+	startupCancel()
+
+	// choose a lan peer and validate lan DHT is functioning.
+	i := rand.Intn(len(lanPeers))
+	if testPeer.PeerHost.Network().Connectedness(lanPeers[i].Identity) == corenet.Connected {
+		i = (i + 1) % len(lanPeers)
+		if testPeer.PeerHost.Network().Connectedness(lanPeers[i].Identity) == corenet.Connected {
+			_ = testPeer.PeerHost.Network().ClosePeer(lanPeers[i].Identity)
+			testPeer.PeerHost.Peerstore().ClearAddrs(lanPeers[i].Identity)
+		}
+	}
+	// That peer will provide a new CID, and we'll validate the test node can find it.
+	provideCid := cid.NewCidV1(cid.Raw, []byte("Lan Provide Record"))
+	provideCtx, cancel := context.WithTimeout(ctx, time.Second)
+	defer cancel()
+	if err := lanPeers[i].DHT.Provide(provideCtx, provideCid, true); err != nil {
+		return err
+	}
+	provChan := testPeer.DHT.FindProvidersAsync(provideCtx, provideCid, 0)
+	prov, ok := <-provChan
+	if !ok || prov.ID == "" {
+		return fmt.Errorf("Expected provider. stream closed early")
+	}
+	if prov.ID != lanPeers[i].Identity {
+		return fmt.Errorf("Unexpected lan peer provided record")
+	}
+
+	// Now, connect with a wan peer.
+	for _, p := range wanPeers {
+		if _, err := mn.LinkPeers(testPeer.Identity, p.Identity); err != nil {
+			return err
+		}
+	}
+
+	err = testPeer.PeerHost.Connect(ctx, wanPeers[0].Peerstore.PeerInfo(wanPeers[0].Identity))
+	if err != nil {
+		return err
+	}
+
+	startupCtx, startupCancel = context.WithTimeout(ctx, time.Second*60)
+WanStartupWait:
+	for {
+		select {
+		case err := <-testPeer.DHT.WAN.RefreshRoutingTable():
+			//if err != nil {
+			//	fmt.Printf("Error refreshing routing table: %v\n", err)
+			//}
+			if testPeer.DHT.WAN.RoutingTable() == nil ||
+				testPeer.DHT.WAN.RoutingTable().Size() == 0 ||
+				err != nil {
+				time.Sleep(100 * time.Millisecond)
+				continue
+			}
+			break WanStartupWait
+		case <-startupCtx.Done():
+			startupCancel()
+			return fmt.Errorf("expected faster wan dht bootstrap")
+		}
+	}
+	startupCancel()
+
+	// choose a wan peer and validate wan DHT is functioning.
+	i = rand.Intn(len(wanPeers))
+	if testPeer.PeerHost.Network().Connectedness(wanPeers[i].Identity) == corenet.Connected {
+		i = (i + 1) % len(wanPeers)
+		if testPeer.PeerHost.Network().Connectedness(wanPeers[i].Identity) == corenet.Connected {
+			_ = testPeer.PeerHost.Network().ClosePeer(wanPeers[i].Identity)
+			testPeer.PeerHost.Peerstore().ClearAddrs(wanPeers[i].Identity)
+		}
+	}
```

Comment:
ditto. Now that we're calling FindProviders instead of FendPeer, disconnecting doesn't buy us anything.